File: atcoder/abc189/b.cpp
int main() {
  int n;
  long double x;
  cin >> n >> x;
  long double a = 0.0;
  int ans = -1;
  rep(i, n) {
    long double v, p;
    cin >> v >> p;
    a += v * p / 100.0;
    if (a > x) {
      ans = i + 1;
      break;
    }
  }
  cout << (ans) << endl;
}

File: atcoder/abc189/b.py
from decimal import *
n, x = map(int, input().split())
x = Decimal(x)
ans = -1
hund = Decimal(100)
acc = Decimal(0)
for i in range(n):
  v, p = map(Decimal, input().split())
  acc += v*p/hund
  if acc > x and ans==-1:
    ans = i+1
print(ans)
